core/domain/change_domain.py

```diff
@@ -223,6 +222,30 @@ def to_dict(self):
 
         return base_change_dict
 
+    @classmethod
+    def from_dict(cls, base_change_dict):
+        """Returns a BaseChange domain object from a dict.
+
+        Args:
+            base_change_dict: dict. The dict representation of
+                BaseChange object.
+
+        Returns:
+            BaseChange. The corresponding BaseChange domain object.
+        """
+        return cls(base_change_dict)
+
+    def validate(self):
+        """Validates various properties of the BaseChange object.
+
+        Raises:
+            ValidationError. One or more attributes of the BaseChange are
+                invalid.
+        """
+
+        base_change_dict = self.to_dict()
+        self.validate_dict(base_change_dict)
```

Comment:
```suggestion
        self.validate_dict(self.to_dict())
```

Also please add an comment why we do this

Reply:
Done